### ISSUE TITLE
[Bug] elasticstack_elasticsearch_security_role_mapping: jsonencode convert list to string if only one item will be passed in the list

### ISSUE THREAD

**Issue by Tim-herbie** [elastic/terraform-provider-elasticstack#457]
**Describe the bug**
When I create a elasticstack_elasticsearch_security_role_mapping resource and pass the field "groups" as a list, the terraform plan and apply will be executed correctly, but it will be stored as string in the .tfstate file instead of a list. The Terraform tries to change this again with every new Plan/Apply

**To Reproduce**
```terraform
resource "elasticstack_elasticsearch_security_role_mapping" "role_mapping" {
  enabled  = true
  name     = each.value.project_name
  roles    = ["kibana_admin"]
  rules = jsonencode({
    all: [
      {
        field: {
          "realm.name": "abc"
        }
      },
      {
        field: {
          groups : ["project1"]
        }
      }
    ]
  })
}
```

**Expected behavior**
The group "project1" will be saved as list in the .tfstate file

**Versions (please complete the following information):**
 - Terraform Version: v1.5.7
 - Provider version: 0.9.0
 - Elasticsearch Version: 8.10.4

**Additional context**
I thought first that this a global terraform problem, but they directed me back to you (https://github.com/hashicorp/terraform/issues/34107)

**Comment by kevn-gonz** [comment]
Correct, this is not expected. We have several role mappings (around 36) and each time we want to make a small update (_to any other terraform resource_), terraform shows there are 36 "changes" to make to the role mappings.
Some of our role mappings are linked to 1 group only, and some others to 2+ groups. Running "terraform apply" shows "changes" only for the role mappings with 1 group linked, not the others.

Looks like terraform takes the role mappings that only have 1 group as strings. so whenever we apply a different change, it detects we're trying to change that string to a list, which is not correct, because during the first run, we applied a list, but for some reason in terraform it is saved as a string.

Attaching some screenshots

![image](https://github.com/elastic/terraform-provider-elasticstack/assets/21992295/3b11d8e6-6d69-4fdf-b722-55bd85d63c67)
![image](https://github.com/elastic/terraform-provider-elasticstack/assets/21992295/baeeacda-a3e9-4afa-8010-79e5b0f09700)

**Comment by rgarcia89** [comment]
This looks more like an issue on elastic side to me. When you check the mapping rules in json editor of kibana you will see that when a list of strings with only one entry is passed it is saved a string. If you however pass a list of strings with multiple values it stays like that.

```json
      "field": {
        "groups": "Azure-K8s-Unicorns-RW"
      }
```

vs
```json
      "field": {
        "groups": [
          "Team-Infrastructure",
          "Team-Network-Operations"
        ]
      }
```

**Comment by eloo-abi** [comment]
can the tf-provider maybe add a workaround here? 
maybe this field is transformed into an array everytime for the compare step?

**Comment by tobio** [comment]
> Should the normalizeRuleNode collapsing behavior on read be removed now that semantic equality handles the discrepancy? Keeping it means state always reflects the canonical ES form (string); removing it would mean state stores whatever the typed client returns (array). Either works with Approach B, but the current behavior is simpler to reason about.

Simplify by removing it. 

> Does the ES API guarantee single-element arrays are always stored as strings, or is this version-dependent behavior? If a future ES version preserves arrays, the normalization would still be harmless but unnecessary.

This is a non-issue either way

> Should the fix be gated to the rolemapping-local package, or does the rule DSL normalization logic belong in a shared utility that other potential rule-bearing resources could reuse?

Keep it local for now, we can move it easily.
